src/plugins/part.c

```diff
@@ -1172,14 +1176,39 @@ gboolean bd_part_resize_part (const gchar *disk, const gchar *part, guint64 size
         return FALSE;
     }
 
+    old_size = ped_part->geom.length * dev->sector_size;
     if (!resize_part (ped_part, dev, ped_disk, size, align, error)) {
         ped_disk_destroy (ped_disk);
         ped_device_destroy (dev);
         bd_utils_report_finished (progress_id, (*error)->message);
         return FALSE;
     }
 
-    ret = disk_commit (ped_disk, disk, error);
+    new_size = ped_part->geom.length * dev->sector_size;
+    if (old_size != new_size) {
+        gint fd = 0;
+        gint wait = 0;
+        guint64 block_size = 0;
+
+        ret = disk_commit (ped_disk, disk, error);
+        /* wait for partition to appear with new size */
+        while (wait < 150) {
```

Comment:
I take it this means 15 seconds? Would be nice to add a comment or something so that people don't have to do the maths every time they look at this code. Plus I think it's a very long timeout for a partition creation.